### PR TITLE
Improve clarity of modules documentation

### PR DESCRIPTION
**Pull request by bfolie**
# Citrine Python PR
PLA-7380

## Description 
A hodgepodge of improvements

- Explains what `check_for_update` is checking for
- Clarify that `GridProcessor` can act along enumerated dimensions
- Connects `ExpressionPredictor` docstring to long-form documentation
- More details around `p` for `MeanPropertyPredictor`
- Explain tradeoffs of changing folds/trials in CV
- Highlight utility of descriptor helper methods
- Throw an error when user tries to trigger a PEW from the wrong collection

### PR Type:
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

### Adherence to team decisions
- [ ] I have added tests for 100% coverage
- [ ] I have written Numpy-style docstrings for every method and class.
- [ ] I have communicated the downstream consequences of the PR to others.
- [ ] I have bumped the version in setup.py

### CODE REVIEW

File: docs/source/workflows/predictors.rst [hunk 1]
```diff
@@ -4,22 +4,28 @@ Predictors
...
 AutoMLPredictors allow you to use your domain knowledge to construct custom `GraphPredictors <#graph-predictor>`__ with fine grain control over the resulting graph.
-Each AutoMLpredictor is defined by a set of inputs and an output.
+Each AutoMLPredictor is defined by a set of inputs and an output.
 Inputs are used as input features to the machine learning model.
 The output is the property that you would like the model to predict.
 There must be at least one input and only one output.
 Unlike the `SimpleMLPredictor <#simple-ml-predictor>`__, only one model is trained from inputs to the output.
 
 Models are trained using data provided by a :class:`~citrine.informatics.data_sources.DataSource` specified when creating a predictor.
+The inputs and outputs are descriptors, which much correspond precisely to descriptors that exist in the training data or are produced by other predictors in the graphical model.
```
Comment (kroenlein) [inline docs/source/workflows/predictors.rst:25]:
```suggestion
The inputs and outputs are descriptors, which must correspond precisely to descriptors that exist in the training data or are produced by other predictors in the graphical model.
```

File: docs/source/workflows/predictors.rst [hunk 2]
```diff
@@ -4,22 +4,28 @@ Predictors
...
+There are two important helper methods in this regard.
+:func:`~citrine.resources.descriptors.DescriptorMethods.descriptors_from_data_source` can provide all of the descriptors that are presesnt in the training data.
```
Comment (kroenlein) [inline docs/source/workflows/predictors.rst:27]:
```suggestion
:func:`~citrine.resources.descriptors.DescriptorMethods.descriptors_from_data_source` can provide all of the descriptors that are present in the training data.
```

File: src/citrine/resources/predictor.py [hunk 3]
```diff
@@ -40,6 +40,9 @@ def check_for_update(self, predictor_id: UUID) -> Optional[Predictor]:
         """
         Check if there are updates available for a predictor.
 
+        Typically these are updates to the training data. For example, a Gem Table may have
```
Comment (kroenlein) [inline src/citrine/resources/predictor.py:43]:
```suggestion
        Typically these are updates to the training data. For example, a GEM Table may have
```

**Review by kroenlein — APPROVED** [review]

**Review by kroenlein — DISMISSED** [review]
There's a bit of trust with this approval, since I'm less versed in this aspect of the stack.

File: src/citrine/resources/delete.py [hunk 4]
```diff
@@ -63,7 +64,7 @@ def _async_gemd_batch_delete(
     scoped_uids = []
     for uid in id_list:  # And now normalize to id/scope pairs
         if isinstance(uid, BaseEntity):
-            link_by_uid = LinkByUID.from_entity(uid)
+            link_by_uid = LinkByUID.from_entity(uid, CITRINE_SCOPE)
```
Comment (kroenlein) [inline src/citrine/resources/delete.py:67]: Good catch

**Review by kroenlein — APPROVED** [review]

File: docs/source/workflows/predictor_evaluation_workflows.rst [hunk 5]
```diff
@@ -26,12 +27,26 @@ Metrics are computed by comparing the model's predictions to observed values.
 Where a metric is computed by taking an average over points in the test folds 
 the fold-wise average is reported as opposed to the point-wise average.
 
-In addition to a name, set of responses to validate, trials, folds and metrics to compute, this evaluator defines a set of descriptor keys to ignore when grouping.
+We recommend 5 folds and 3 trials as the default.
```
Comment (talltea) [inline docs/source/workflows/predictor_evaluation_workflows.rst:30]: this is super helpful for me! thanks for adding this discussion 👍

**Review by talltea — DISMISSED** [review]
👍 lgtm

**Review by talltea — APPROVED** [review]

**Pull request merged**